packages/tldraw/src/lib/defaultExternalContentHandlers.ts

```diff
@@ -81,19 +82,63 @@ export function registerDefaultExternalContentHandlers(
 			}
 		}
 
-		// Always rescale the image
-		if (file.type === 'image/jpeg' || file.type === 'image/png') {
-			file = await downsizeImage(file, size.w, size.h, {
-				type: file.type,
-				quality: 0.92,
-			})
-		}
-
 		const assetId: TLAssetId = AssetRecordType.createId(hash)
 
-		const asset = AssetRecordType.create({
+		if (isImageType) {
+			const sources: TLImageAsset['props']['sources'] = []
+
+			// Always rescale the image
+			if (file.type === 'image/jpeg' || file.type === 'image/png') {
+				sources.push(
+					{
+						scale: 1,
+						src: await FileHelpers.blobToDataUrl(
+							await downsizeImage(file, size.w, size.h, {
```

Comment:
ditto re: doing some of this down-sizing in parallel

Reply:
hah, i have a similar concern here on parallelization with less powerful machines and large images :)